packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1153,6 +1167,15 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
         ),
     ];
 
+    // Ensure configuration uses Material text style for misspelled words unless
+    // a custom style is specified.
+    final SpellCheckConfiguration? spellCheckConfiguration =
+      widget.spellCheckConfiguration != null
+        ? widget.spellCheckConfiguration!.copyWith(
+            misspelledTextStyle: widget.spellCheckConfiguration!.misspelledTextStyle
+              ?? TextField.materialMisspelledTextStyle)
+        : null;
```

Comment:
So currently, the behavior is the same if spellCheckConfiguration is explicitly passed `null` and if it is not specified at all.  Are we still planning to enable spell check by default at some point, and if so, would it be possible to distinguish between passed `null` and not passed?

Reply:
To update this as per our offline discussion on the discussion of the meaning of various values for the `spellCheckConfiguration` parameters of `TextField`s and `EditableText`:

#### When Spell Check Disabled by Default (now)

- Passing `null` means to disable spell check.
- **[`TextField` only]** Passing `SpellCheckConfiguration()` means defaults are wanted for `SpellCheckService`, `SpellCheckSuggestionsHandler`, and misspelled `TextStyle`.
- **[`EditableText` only]**  Passing `SpellCheckConfiguration(misspelledTextStyle: someStyle)` means defaults are wanted for `SpellCheckService` and `SpellCheckSuggestionsHandler`.

#### When Spell Check Enabled by Default (later)

- Passing `null` means to enable spell check.
- Passing `SpellCheckConfiguration()` means to disable spell check. This will be defined as a constant in the `SpellCheckConfiguration` class (something like `SpellCheckConfiguration.disabled`).
- Passing `SpellCheckConfiguration(misspelledTextStyle: someStyle)` means defaults are wanted for `SpellCheckService` and `SpellCheckSuggestionsHandler`.